.gitlab-ci.yml

```
@@ -193,6 +193,27 @@ cargo-check-subkey:
     - BUILD_DUMMY_WASM_BINARY=1 time cargo check --release
     - sccache -s
 
+test-deterministic-wasm:           &test-linux
+  stage:                           test
+  <<:                              *docker-env
+  variables:
+    <<:                            *default-vars
+  except:
+    variables:
+      - $DEPLOY_TAG
+  script:
+    # build runtime
+    - WASM_BUILD_NO_COLOR=1 cargo build --verbose --release -p node-runtime
+    # make checksum
+    - sha256sum target/release/wbuild/target/wasm32-unknown-unknown/release/node_runtime.wasm > checksum.sha256
+    # clean up – FIXME: can we reuse some of the artifacts?
+    - cargo clean
+    # build again
+    - WASM_BUILD_NO_COLOR=1 cargo build --verbose --release -p node-runtime
+    # confirm checksum
+    - sha256sum -c checksum.sha256
+
```

```suggestion
   - sccache -s
```

Useful to have `sccache -s` after jobs, for occasional debugging of jobs.

---

should that maybe happen before the checksum then? Because if the checksum fails, this is aborted ... but for debugging that is the interesting case, no?